src/Visualization/Python/Render1D.py

```diff
@@ -0,0 +1,279 @@
+    volfile = h5files[0]["element_data.vol"]
+    obs_id_0 = next(iter(volfile))
+    variables = list(volfile[obs_id_0].keys())
+    variables.remove("connectivity")
+    variables.remove("InertialCoordinates_x")
+    variables.remove("total_extents")
+    variables.remove("grid_names")
+    print("Variables in H5 file:\n{}".format(list(map(str, variables))))
+    for h5_file in h5files:
+        h5_file.close()
+    return None
+
+
+def get_data(files, var_name):
+    '''
+    Get the data to be plotted
+    :return: the array of data, coords, and time
+    '''
+
+    time = []
+    coords = []
+    data = []
+    h5files = get_h5_files(files)
+    volfiles = [h5file['element_data.vol'] for h5file in h5files]
+    # Get a list of times from the first vol file
+    ids_times = [(obs_id, volfiles[0][obs_id].attrs['observation_value'])
+                 for obs_id in volfiles[0].keys()]
+    ids_times.sort(key=lambda pair: pair[1])
+    for obs_id, local_time in ids_times:
+        local_coords = []
+        local_data = []
+        ordering = None
+        for volfile in volfiles:
+            try:
+                local_data = local_data +\
+                             list(volfile[obs_id][var_name])
+            except KeyError:
+                print("The variable name {} is not a valid variable. "
+                      "Use '--list-vars' to see the list of variable names in "
+                      "the file(s) \n{}"\
+                      .format(var_name, list(map(str, h5files))))
+                sys.exit(1)
+            local_coords = local_coords + \
+            list(volfile[obs_id]['InertialCoordinates_x'])
+        ordering = np.argsort(local_coords)
+        coords.append(np.array(local_coords)[ordering])
+        data.append(np.array(local_data)[ordering])
+        time.append(local_time)
+
+    for h5file in h5files:
+        h5file.close()
+    return time, coords, data
+
+
+def render_single_time(var_name, time_slice, output_prefix, coords, data):
+    '''
+    Renders a single time snap shot
+    :param args: the command line arguments
+    :return: None
+    '''
+    import matplotlib.pyplot as plt
+
+    plt.xlabel("x")
+    plt.ylabel(var_name)
+    plt.title("t = {:.5f}".format(int(time_slice)))
+    plt.plot(coords[int(time_slice)], data[int(time_slice)], 'o')
+    if output_prefix:
+        logging.info("Writing still to file {}.pdf".format(output_prefix))
+        plt.savefig(output_prefix + ".pdf", format='pdf')
+    else:
+        plt.show()
+    return None
+
+
+def render_animation(var_name, output_prefix, my_interval, time, coords, data):
+    '''
+    Render an animation of the data
+    :param args: the command line arguments
+    :return: None
+    '''
+    import matplotlib.pyplot as plt
+    import matplotlib.animation as animation
+
+    def init():
+        '''
+        Initialize the animation canvas
+        :return: empty line and title info
+        '''
+        line.set_data([], [])
```

Comment:
before this line add:
```py
global line, title
```
(same in the `animate` function below)

Reply:
When I add these lines I get the following error:
`NameError: global name 'line' is not defined `

Reply:
you should move the commands where you initialize `line` and `title` up, so they run before the `init` and `animate` functions are defined. I add a comment to the relevant code change below.

Reply:
This still gives me the `NameError` probably because `line`, `title` are initialized in the `render_animation` function and not globally. I could also try `nonlocal` but this is different in Python 2 and 3. Is there anything else I can try?

Reply:
You're right that `nonlocal` would be the correct choice, but it doesn't exist in Python 2. Let's leave it as-is then, without a `global` or `nonlocal` statement. As long as you don't assign to the `line` or `title` variables but only call member functions on them the code should be fine. You can delete lines 206 and 207.